rtgui/edit.h

```diff
@@ -152,6 +145,59 @@ class EditBuffer;
  *
  */
 
+
+
+class ObjectMOBuffer
+{
+private:
+
+    // Used to draw the objects where the color correspond to the object's ID, in order to find the correct object when hovering
```

Comment:
I think whether we should use a `contains` method is definitely out of scope for this change here and I guess we should not decide on it without trying it out first. As you mention, the polygon case could become complicated even though standard algorithms exist. The usage of `FORMAT_RGB16_565` might still be a sensible change and performance improvement.

In the long run, I'd say this really looks like it wants to be a general purpose scene graph library, c.f. [QGraphcisScene](https://doc.qt.io/qt-5/qgraphicsscene.html) or [Papyrus](http://libpapyrus.sourceforge.net/index.html). So we should probably look into working with an existing library instead of rolling our own.